newIDE/app/src/ExpressionAutocompletion/index.js

```diff
@@ -207,17 +220,77 @@ const getAutocompletionsForObject = function(
     ...filteredObjectsList.map(({ object }) => ({
       kind: 'Object',
       completion: object.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       object,
       addDot,
     })),
     ...filteredGroupsList.map(({ group }) => ({
       kind: 'Object',
       completion: group.getName(),
+      replacementStartPosition: completionDescription.getReplacementStartPosition(),
+      replacementEndPosition: completionDescription.getReplacementEndPosition(),
       addDot,
     })),
   ];
 };
 
+const getAutocompletionsForText = function(
+  expressionAutocompletionContext: ExpressionAutocompletionContext,
+  completionDescription: gdExpressionCompletionDescription
+): Array<ExpressionAutocompletion> {
+  const prefix: string = completionDescription.getPrefix();
+  const type: string = completionDescription.getType();
+  //const objectName: string = completionDescription.getObjectName();
+  const {
+    globalObjectsContainer,
+    objectsContainer,
+  } = expressionAutocompletionContext;
+
+  let autocompletionTexts: string[] = [];
+  if (type === 'layer') {
+    const layout = (objectsContainer: gdLayout);
+    for (let index = 0; index < layout.getLayersCount(); index++) {
+      autocompletionTexts.push(`"${layout.getLayerAt(index).getName()}"`);
+    }
+  } else if (type === 'sceneName') {
+    const project = (globalObjectsContainer: gdProject);
```

Comment:
We should be able to pass the `project` in the `expressionAutocompletionContext` when calling `getAutocompletionsFromDescriptions` :)